test/worldwide/region_yml_consistency_test.rb

```diff
@@ -100,6 +100,35 @@ class RegionYmlConsistencyTest < ActiveSupport::TestCase
       end
     end
 
+    test "additioanl_address_field names must belong to a limited set of allowed names" do
+      allowed_names = ["streetName", "streetNumber", "line2", "neighborhood"]
+
+      Regions.all.select(&:country?).each do |country|
+        next if country.additional_address_fields.blank?
+
+        country.additional_address_fields.each do |field|
+          field_name = field["name"]
+
+          assert_includes allowed_names, field_name, "#{country.iso_code} additional_address_field #{field_name} is not an allowed name"
+        end
+      end
+    end
+
+    test "additioanl_address_field names are present as keys in the combined_address_format" do
+      Regions.all.select(&:country?).each do |country|
+        next if country.additional_address_fields.blank?
+
+        country.additional_address_fields.each do |field|
+          field_name = field["name"]
+          present = country.combined_address_format.any? do |_, fields|
+            fields.any? { |f| f["key"] == field_name }
+          end
+
+          assert present, "#{country.iso_code} additional_address_field #{field_name} is not present in combined_address_format"
+        end
+      end
+    end
+
```

Comment:
Can we also add a test to make sure all keys in `combined_address_format` are also in `additional_address_fields`?